### PR TITLE
Renggli et al. implementation

### LINKED ISSUE #4: Implement Renggli et al
[Cedric Renggli, André Susano Pinto, Luka Rimanic, Joan Puigcerver, Carlos Riquelme, Ce Zhang, and
Mario Lučić. Which model to transfer? finding the needle in the growing haystack. In Proceedings of the
IEEE/CVF Conference on Computer Vision and Pattern Recognition, pages 9205–9214, 2022](https://openaccess.thecvf.com/content/CVPR2022/papers/Renggli_Which_Model_To_Transfer_Finding_the_Needle_in_the_Growing_CVPR_2022_paper.pdf)

**Task-aware strategy:**

- Train a logistic regression model on (a subset of) the new dataset, i.e. remove the last layer, do a forward pass to get features from the pre-trained model, train model on those features.

- Optionally: Same but with KNN classifier (but scales worse than above and performance was the same/slightly worse in the paper)

**Task-agnostic strategy: [NB: will save implementing this for #10]**

In this order:

1. Pick the model with the highest test accuracy on ImageNet.
2. Pick the model trained on the largest dataset.
3. Pick the model with the most parameters.

### PR DESCRIPTION
**Pull request by jack89roberts**
See notebook here:

- https://github.com/alan-turing-institute/ARC-LoCoMoSeT/blob/4-implement-renggli/notebooks/renggli_dev.ipynb

Also adds actions & pre-commit for notebooks.

TODO:

- [x] Get it working on different models
- [x] Get it working on larger batches (i.e. figure out how to run it on all of/a big chunk of ImageNet)
- [x] Pull out code into scripts/package
- [x] Add tests

Will be a lot of overlap between this, #6, and other metrics on the data processing/feature extracting (may split that into separate PRs before merging this one). 

Closes #4

### CODE REVIEW

**Comment by jack89roberts** [comment]
Merging this one then will consolidate on the [milestone/wp1](https://github.com/alan-turing-institute/ARC-LoCoMoSeT/tree/milestone/wp1) branch.
